pandas/tests/io/test_iceberg.py

```diff
@@ -0,0 +1,148 @@
+"""
+Tests for the Apache Iceberg format.
+
+Tests in this file use a simple Iceberg catalog based on SQLite, with the same
+data used for Parquet tests (``pandas/tests/io/data/parquet/simple.parquet``).
+"""
+
+from contextlib import contextmanager
+import importlib
+import pathlib
+import tempfile
+
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
+from pandas.io.iceberg import read_iceberg
+
+pytestmark = [pytest.mark.single_cpu]
+
+pyiceberg = pytest.importorskip("pyiceberg")
+pyiceberg_catalog = pytest.importorskip("pyiceberg.catalog")
+pq = pytest.importorskip("pyarrow.parquet")
+
+
+@contextmanager
+def create_catalog(catalog_name_in_pyiceberg_config=None):
```

Comment:
Does it make sense for this to be a pytest fixture? It could be parametrized over different catalog names by default and could use the `tmpdir` fixture to do the temporary directory stuff automatically

Reply:
That was my first implementation, but I couldn't make a fixture remove the files when the test finish. That's why I implemented it as a context manager.

I'll check again, as I didn't try with the tmpdir fixture, but it may not be easy.